node/node.go

```diff
@@ -631,27 +634,27 @@ func (node *Node) UpdateStakingList(block *types.Block) error {
 	txns := block.Transactions()
 	for i := range txns {
 		txn := txns[i]
-		value := txn.Value().Int64()
-		currentSender, _ := types.Sender(signerType, txn)
-		_, isPresent := node.CurrentStakes[currentSender]
 		toAddress := txn.To()
 		if *toAddress != node.StakingContractAddress { //Not a address aimed at the staking contract.
 			continue
 		}
-		//This should be based on a switch case on function signature.
-		//TODO (ak) https://github.com/harmony-one/harmony/issues/430
-		if value > int64(0) { //If value >0 means its a staking deposit transaction
+		currentSender, _ := types.Sender(signerType, txn)
+		_, isPresent := node.CurrentStakes[currentSender]
+		data := txn.Data()
+		switch funcSignature := decodeFuncSign(data); funcSignature {
+		case "0xd0e30db0": //deposit
+			amount := txn.Value()
+			value := amount.Int64()
 			if isPresent {
-				//This means this node has increaserd its stake
+				//This means the node has increased its stake.
 				node.CurrentStakes[currentSender] += value
 			} else {
+				//This means its a new node that is staking the first time.
 				node.CurrentStakes[currentSender] = value
 			}
-		} else { //This means node has withdrawn stake.
-			getData := txn.Data()
-			value := decodeStakeCall(getData) //Value being withdrawn
+		case "0x2e1a7d4d": //withdaw
```

Comment:
constant - please comment where did you get the constant.

Reply:
Done.